src/core/net/ip6.cpp

```diff
@@ -426,6 +427,11 @@ ThreadError Ip6::HandleDatagram(Message &message, Netif *netif, int8_t interface
     {
         if (netif != NULL)
         {
+            if (otGetDeviceRole() == kDeviceRoleChild)
```

Comment:
That's interesting behavior.  An IPv6 message should only be forwarded to the SED if the IPv6 destination address is registered by the SED.  In what case does the parent forward a message to the SED and the IPv6 destination address does not belong to the SED?